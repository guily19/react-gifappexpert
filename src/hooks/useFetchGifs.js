import { useState, useEffect } from 'react';
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( category ) => {
  const [state, setState] = useState({
    data: [],
    loading: true
  });

  useEffect(() => {
    getGifs( category ).then( imgs => {
      setState({
        data: imgs,
        loading: false
      });
    })
  }, [category])
  console.log('returning state:', state);
  return state;
};
