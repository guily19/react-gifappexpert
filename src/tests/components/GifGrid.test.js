import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('<GifGrid /> tests', () => {
  const category = 'test category';

  test('<GifGrid /> renders correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={category}/>);
    expect(wrapper).toMatchSnapshot();
  });

  test('<GifGrid /> renders correctly with data', () => {
    const gifs = [{
      title: 'image title',
      url: 'http://testurl.com',
      id: 'ABC'
    }];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={category}/>);
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
  });
});