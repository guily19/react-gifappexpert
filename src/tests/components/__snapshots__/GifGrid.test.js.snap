// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> tests <GifGrid /> renders correctly 1`] = `
<Fragment>
  <h3>
     
    test category
     
  </h3>
  <p>
     Loading.. 
  </p>
  <div
    className="card-grid animate__animated animate__heartBeat"
  />
</Fragment>
`;
